test/e2e-test/appconfig_dependency_test.go

```diff
@@ -0,0 +1,414 @@
+		Expect(k8sClient.Delete(ctx, &wd)).Should(BeNil())
+		Expect(k8sClient.Delete(ctx, &td)).Should(BeNil())
+		// delete the namespace with all its resources
+		Expect(k8sClient.Delete(ctx, &ns, client.PropagationPolicy(metav1.DeletePropagationForeground))).Should(BeNil())
+	})
+
+	// common function for verification
+	verify := func(appConfigName string) {
+		// Verification before satisfying dependency
+		By("Checking that resource which accepts data isn't created yet")
+		inFooKey := client.ObjectKey{
+			Name:      inName,
+			Namespace: namespace,
+		}
+		inFoo := tempFoo.DeepCopy()
+		logf.Log.Info("Checking on resource that inputs data", "Key", inFooKey)
+		Eventually(
+			func() error {
+				return k8sClient.Get(ctx, inFooKey, inFoo)
+			},
+			time.Second*15, time.Millisecond*500).Should(&util.NotFoundMatcher{})
+		By("Checking that resource which provides data is created")
+		outFooKey := client.ObjectKey{
+			Name:      outName,
+			Namespace: namespace,
+		}
+		outFoo := tempFoo.DeepCopy()
+		logf.Log.Info("Checking on resource that outputs data", "Key", outFoo)
+		Eventually(
+			func() error {
+				return k8sClient.Get(ctx, outFooKey, outFoo)
+			},
+			time.Second*15, time.Millisecond*500).Should(BeNil())
+		By("Verify the appconfig's dependency is unsatisfied")
+		appconfigKey := client.ObjectKey{
+			Name:      appConfigName,
+			Namespace: namespace,
+		}
+		appconfig := &v1alpha2.ApplicationConfiguration{}
+		depStatus := v1alpha2.DependencyStatus{
+			Unsatisfied: []v1alpha2.UnstaifiedDependency{
+				{
+					From: v1alpha2.DependencyFromObject{
+						TypedReference: v1alpha1.TypedReference{
+							APIVersion: tempFoo.GetAPIVersion(),
+							Name:       outName,
+							Kind:       tempFoo.GetKind(),
+						},
+						FieldPath: "status.key",
+					},
+					To: v1alpha2.DependencyToObject{
+						TypedReference: v1alpha1.TypedReference{
+							APIVersion: tempFoo.GetAPIVersion(),
+							Name:       inName,
+							Kind:       tempFoo.GetKind(),
+						},
+						FieldPaths: []string{
+							"spec.key",
+						},
+					},
+				},
+			},
+		}
+		logf.Log.Info("Checking on appconfig", "Key", appconfigKey)
+		Eventually(
+			func() v1alpha2.DependencyStatus {
+				k8sClient.Get(ctx, appconfigKey, appconfig)
+				return appconfig.Status.Dependency
+			},
+			time.Second*15, time.Millisecond*500).Should(Equal(depStatus))
+		// fill value to fieldPath
+		err := unstructured.SetNestedField(outFoo.Object, "test", "status", "key")
+		Expect(err).Should(BeNil())
+		Expect(k8sClient.Update(ctx, outFoo)).Should(Succeed())
+		// Verification after satisfying dependency
+		By("Checking that resource which accepts data is created now")
+		logf.Log.Info("Checking on resource that inputs data", "Key", inFooKey)
+		Eventually(
+			func() error {
+				return k8sClient.Get(ctx, inFooKey, inFoo)
+			},
+			time.Second*15, time.Millisecond*500).Should(BeNil())
+		By("Verify the appconfig's dependency is unsatisfied")
```

Comment:
satisfied?

Reply:
I have fixed the logging error now. @ryanzhang-oss